### PR TITLE
40percentclub 4 pack macropad keyboard

### PR DESCRIPTION
**Pull request by Ardakilic**
4 Pack
--------

4pack

A 4 Key macropad PCB

You can find the main blog post about this macropad [here](http://www.40percent.club/2017/07/4-pack.html), also the Gerber files and other documentation is located [here](https://git.40percent.club/di0ib/Misc/src/branch/master/4pack).

I've realized this keyboard was missing a QMK firmware, unlike other 40percentclub PCBs, so I've decided to make one to benefit from QMK.

There were two struggles while I was building this keyboard firmware:

1. The pinout didn't have a matrix, pins were directly connected to the io pins. I've realized for the row value, I needed to use ~`B0` (ground pin)~ direct pins instead of matrix in this case.
2. Leds' both anode and cathode pins were connected to io pins, so some GPIO commands needed to set them and light them.

While building, I've used header pins directly instead of sockets as mentioned, also, I've used 1k ohm resistors, along with Gateron greens (my favorite cheap macropad switches). These worked out nicely.

### CODE REVIEW

File: keyboards/40percentclub/4pack/keymaps/default/readme.md [hunk 1]
```diff
@@ -0,0 +1 @@
+# The default keymap for %KEYBOARD%
```
Comment (noroadsleft) [inline keyboards/40percentclub/4pack/keymaps/default/readme.md]:
```suggestion
# The default keymap for 4pack
```

File: keyboards/40percentclub/4pack/readme.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+I've realized this keyboard was missing a QMK firmware, so I've decided to make one to benefit from QMK.
+
+There were two struggles while I was building this keyboard firmware:
+
+1. The pinout didn't have a matrix, pins were directly connected to the io pins. I've realized for the row value, I needed to use `B0` (ground pin) in this case.
+2. Leds' both anode and cathode pins were connected to io pins, so some GPIO commands needed to set them and light them.
+
+While building, I've used header pins directly instead of sockets as mentioned, also, I've used 1k ohm resistors, along with Gateron greens (my favorite cheap macropad switches). These worked out nicely.
+
+Keyboard Maintainer: [40percentclub](http://www.40percent.club/)
```
Comment (noroadsleft) [inline keyboards/40percentclub/4pack/readme.md]:
```suggestion
```

File: keyboards/40percentclub/4pack/readme.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# 4 Pack
+
+![4pack](https://i.imgur.com/rioXXq5l.jpg)
+
+A 4 Key macropad PCB
```
Comment (noroadsleft) [inline keyboards/40percentclub/4pack/readme.md]:
```suggestion
A 4-key macropad PCB with its switch and LED pins wired directly to microcontroller IO pins.
```

File: keyboards/40percentclub/4pack/readme.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+I've realized this keyboard was missing a QMK firmware, so I've decided to make one to benefit from QMK.
+
+There were two struggles while I was building this keyboard firmware:
+
+1. The pinout didn't have a matrix, pins were directly connected to the io pins. I've realized for the row value, I needed to use `B0` (ground pin) in this case.
+2. Leds' both anode and cathode pins were connected to io pins, so some GPIO commands needed to set them and light them.
+
+While building, I've used header pins directly instead of sockets as mentioned, also, I've used 1k ohm resistors, along with Gateron greens (my favorite cheap macropad switches). These worked out nicely.
+
```
Comment (noroadsleft) [inline keyboards/40percentclub/4pack/readme.md]: Lines 9-17 should be deleted.

File: keyboards/40percentclub/4pack/readme.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+Keyboard Maintainer: [40percentclub](http://www.40percent.club/)
+Firmware Build: [Arda Kilicdagi](http://github.com/ardakilic)
```
Comment (noroadsleft) [inline keyboards/40percentclub/4pack/readme.md]:
```suggestion
Keyboard Maintainer: [Arda Kilicdagi](http://github.com/ardakilic), [The QMK Community](https://github.com/qmk)  
```

File: keyboards/40percentclub/4pack/readme.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+Hardware Availability: links to where you can find this hardware
```
Comment (noroadsleft) [inline keyboards/40percentclub/4pack/readme.md]:
```suggestion
Hardware Supported: 4 Pack PCB, Pro Micro  
Hardware Availability: [40percent.club](https://git.40percent.club/di0ib/Misc/src/branch/master/4pack)
```

File: keyboards/40percentclub/4pack/rules.mk [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+# If you don't know the bootloader type, then you can specify the
+# Boot Section Size in *bytes* by uncommenting out the OPT_DEFS line
+#   Teensy halfKay      512
+#   Teensy++ halfKay    1024
+#   Atmel DFU loader    4096
+#   LUFA bootloader     4096
+#   USBaspLoader        2048
+# OPT_DEFS += -DBOOTLOADER_SIZE=4096
+
```
Comment (noroadsleft) [inline keyboards/40percentclub/4pack/rules.mk]: Lines 14-22 should be deleted.

File: keyboards/40percentclub/4pack/rules.mk [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support (+2400 to 4200, depending on config)
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no 		# Enable support for HD44780 based LCDs (+400)
```
Comment (noroadsleft) [inline keyboards/40percentclub/4pack/rules.mk]: Lines 26-42 should be:

```Makefile
BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
MIDI_ENABLE = no            # MIDI support
UNICODE_ENABLE = no         # Unicode
BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
AUDIO_ENABLE = no           # Audio output on port C6
FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
```

This removes the firmware impact numbers (they're basically never accurate) and aligns the comments.

**Comment by Ardakilic** [comment]
@zvecr Thanks for the suggestions! To be honest, I just learned about the direct pin definition. I'm curently trying with direct pins definitions. If all goes well, I'll update this pull request.

@noroadsleft Thanks for the review, all should be done unless I missed something on the way.

File: keyboards/40percentclub/4pack/readme.md [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+
+You can find the main blog post about this macropad [here](http://www.40percent.club/2017/07/4-pack.html), also the Gerber files and other documentation is located [here](https://git.40percent.club/di0ib/Misc/src/branch/master/4pack).
+
+Keyboard Maintainer: [Arda Kilicdagi](http://github.com/ardakilic), [The QMK Community](https://github.com/qmk)  
+Hardware Supported: 4 Pack PCB, Pro Micro  
+Hardware Availability: [40percent.club](https://git.40percent.club/di0ib/Misc/src/branch/master/4pack)
+
+Make example for this macropad (after setting up your build environment):
+
+    make 40percentclub/4pack:default:avrdude
```
Comment (drashna) [inline keyboards/40percentclub/4pack/readme.md]: We have a universal flashing mechanic now, that reads the config and will auto-detect which bootloader it's using.  

```suggestion
    make 40percentclub/4pack:default:flash
```

Comment (Ardakilic) [reply keyboards/40percentclub/4pack/readme.md]: Updated in my last commit, thanks for the heads-up.

**Comment by drashna** [comment]
Thanks!

**Comment by Ardakilic** [comment]
Awesome, thanks again for the reviews and the merge!

**Pull request merged**